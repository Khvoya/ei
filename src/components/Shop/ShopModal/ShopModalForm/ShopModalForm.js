import { Button, Form } from "react-bootstrap";
import React, { Component } from "react";
import { sendOrderMail } from "actionCreators/actionCreators";
import PropTypes from "prop-types";
import { connect } from "react-redux";

class ShopModalForm extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    onHide: PropTypes.func.isRequired,
    price: PropTypes.string.isRequired
  };

  state = { validated: false };

  handleSubmit = (event) => {
    const { name, price, onHide } = this.props;
    const { email, comment } = this.state;
    const isValid = event.currentTarget.checkValidity();
    event.preventDefault();
    event.stopPropagation();
    this.setState({ validated: true });
    if (!isValid) return;

    this.props.dispatch(sendOrderMail({
      name,
      price,
      email,
      comment,
    }));
    onHide();
  };

  setClientEmail = e => {
    this.setState({ email: e.target.value });
  };

  setClientComment = e => {
    this.setState({ comment: e.target.value });
  };

  render() {
    const { validated } = this.state;
    return (
      <Form
        noValidate
        validated={validated}
        onSubmit={this.handleSubmit}
      >
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control
            required
            onChange={this.setClientEmail}
            type="email"
            placeholder="Enter email"
          />
          <Form.Text className="text-muted" />
        </Form.Group>
        <Form.Group controlId="formComment">
          <Form.Label>Write some comment here</Form.Label>
          <Form.Control
            onChange={this.setClientComment}
            as="textarea"
            rows="3"
            placeholder="Any additional info (e.g. share your social profile or clarify something)"
          />
        </Form.Group>
        <Button variant="dark" type="submit" >
          Order item
        </Button>
      </Form>
    );
  }
}
export default connect()(ShopModalForm);
