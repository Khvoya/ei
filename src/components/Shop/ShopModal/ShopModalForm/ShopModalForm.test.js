import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ShopModalForm from './ShopModalForm';

describe('ShopModalForm', () => {

  it('should render correctly', () => {
    const output = shallow(
      <ShopModalForm />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
