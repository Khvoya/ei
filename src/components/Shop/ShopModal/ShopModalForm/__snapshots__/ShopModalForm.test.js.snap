// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShopModalForm should render correctly 1`] = `
<Form
  as="form"
  inline={false}
  noValidate={true}
  onSubmit={[Function]}
  validated={false}
>
  <FormGroup
    as="div"
    controlId="formBasicEmail"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Email address
    </FormLabel>
    <FormControl
      as="input"
      onChange={[Function]}
      placeholder="Enter email"
      required={true}
      type="email"
    />
    <FormText
      as="small"
      className="text-muted"
    />
  </FormGroup>
  <FormGroup
    as="div"
    controlId="formComment"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Write some comment here
    </FormLabel>
    <FormControl
      as="textarea"
      onChange={[Function]}
      placeholder="Any additional info (e.g. share your social profile or clarify something)"
      rows="3"
    />
  </FormGroup>
  <Button
    active={false}
    disabled={false}
    type="submit"
    variant="dark"
  >
    Order item
  </Button>
</Form>
`;
